src/server/service.rs

```diff
@@ -0,0 +1,102 @@
+use std::io;
+use std::rc::Rc;
+use std::sync::Arc;
+
+pub trait Service {
+    /// Requests handled by the service.
+    type Request;
+
+    /// Responses given by the service.
+    type Response;
+
+    /// Errors produced by the service.
+    type Error;
+
+    /// Process the request and return the response asynchronously.
```

Comment:
"asynchronously"?

Reply:
https://en.wiktionary.org/wiki/asynchronously

Reply:
The invocation of this function is synchronous or did I miss anything?

Reply:
Oh, sorry, I wasn't seeing the function declaration that is below your comment.

Reply:
Fixed! Now it is asynchronous again :)

Reply:
Originally I thought that it should not - all the usecases I can find will use `type Future = Ready<..>`. But its an unnecessary API breaking change, so lets keep it async?